### ISSUE TITLE
Better way to test requests with MockJsonResources.

### ISSUE THREAD

**Issue by criske** [self-xdsd/self-core#389]
Right now the only way to test requests is in the lambda passed to MockJsonResources' constuctor.
But that sometimes could make tests hard to follow; for example: https://github.com/self-xdsd/self-core/blob/1f16aae71a27c703630f1342c2a4e3cc40bb58a7/self-core-impl/src/test/java/com/selfxdsd/core/GitlabWebhooksTestCase.java#L51

This issue could be fixed if `MockJsonResources` could track all the requests. The requests history is just an iterable with some convenience methods like `first()`,`last()` and `atIndex(int)` .
Now  `GitlabWebhooksTestCase#addsWebhookCreated()` would be shorter and concise:
```java
final MockJsonResources resources = new MockJsonResources(
            new AccessToken.Gitlab("gitlab123"),
            req -> MockResource(HttpURLConnection.HTTP_CREATED, JsonValue.NULL));
//.... 

final boolean res = provider.repo("amihaiemil", "repo") .webhooks().add(project);
final MockRequest req = resources.requests().last();

MatcherAssert.assertThat(req.getAccessToken().value(), Matchers.equalTo("gitlab123"));
MatcherAssert.assertThat(req.getMethod(),Matchers.equalTo("POST"));
//... other asserts
```
I'll do a quick PR. No need for bounty on this one .

**Comment by amihaiemil** [comment]
@criske thanks!
